crates/router/src/connector/paypal/transformers.rs

```diff
@@ -0,0 +1,490 @@
+use masking::Secret;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    connector::utils::{
+        AccessTokenRequestInfo, CardData, PaymentsAuthorizeRequestData, PaymentsRequestData,
+    },
+    core::errors,
+    pii,
+    types::{self, api, storage::enums as storage_enums},
+};
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum PaypalPaymentIntent {
+    Capture,
+    Authorize,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Eq, PartialEq, Deserialize)]
+pub struct OrderAmount {
+    currency_code: String,
+    value: String,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct PurchaseUnitRequest {
+    reference_id: String,
+    amount: OrderAmount,
+}
+
+#[derive(Debug, Serialize)]
+pub struct Address {
+    address_line_1: Option<Secret<String>>,
+    postal_code: Option<Secret<String>>,
+    country_code: String,
+}
+
+#[derive(Debug, Serialize)]
+pub struct CardRequest {
+    billing_address: Option<Address>,
+    expiry: Option<Secret<String>>,
+    name: Secret<String>,
+    number: Option<Secret<String, pii::CardNumber>>,
+    security_code: Option<Secret<String>>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum PaymentSourceItem {
+    Card(CardRequest),
+}
+
+#[derive(Debug, Serialize)]
+pub struct PaypalPaymentsRequest {
+    intent: PaypalPaymentIntent,
+    purchase_units: Vec<PurchaseUnitRequest>,
+    payment_source: Option<PaymentSourceItem>,
+}
+
+fn get_address_info(address: Option<&api_models::payments::Address>) -> Option<Address> {
+    address.and_then(|add| {
+        add.address.as_ref().map(|a| Address {
+            country_code: a.country.clone().unwrap_or_default(),
+            address_line_1: a.line1.clone(),
+            postal_code: a.zip.clone(),
+        })
+    })
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for PaypalPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        match item.request.payment_method_data.clone() {
+            api_models::payments::PaymentMethodData::Card(ref ccard) => {
+                let intent = match item.request.is_auto_capture() {
+                    true => PaypalPaymentIntent::Capture,
+                    false => PaypalPaymentIntent::Authorize,
+                };
+                let amount = OrderAmount {
+                    currency_code: item.request.currency.to_string().to_uppercase(),
+                    value: item.request.amount.to_string(),
+                };
+                let reference_id = item.attempt_id.clone();
+
+                let purchase_units = vec![PurchaseUnitRequest {
+                    reference_id,
+                    amount,
+                }];
+                let card = item.get_card()?;
+                let expiry = Some(card.get_expiry_date_as_yyyymm("-")).map(Secret::new);
```

Comment:
shouldn't the get_expiry_date_as_yyyymm method return `Secret` always. Because it's a PII data